src/graph/paths.cc

```diff
@@ -452,8 +455,11 @@ ncclResult_t ncclTopoNeedFlush(struct ncclTopoSystem* system, int64_t busId, int
   int g;
   NCCLCHECK(ncclTopoIdToIndex(system, GPU, busId, &g));
   struct ncclTopoNode* gpu = system->nodes[GPU].nodes+g;
+#if defined(__HIP_PLATFORM_AMD__) || defined(__HCC__) || defined(__HIPCC__)
+#else
```

Comment:
So, on AMD platform, we will not flush the data because it is not required.

Reply:
yes

Reply:
Does it change anything in terms of performance?

Reply:
I think flush incurs a few microseconds latency